File: MythonJava/src/sample/types/Void.java
package sample.types;

public class Void extends Type {
    @Override
    public boolean isPrintable() {
        return false;
    }

    @Override
    public String toString() {
        return null;
    }

    @Override
    public String getType() {
        return "<type 'void'>";
    }
}

File: MythonC++/Mython/Void.cpp
#include "Void.h"

Void::Void() : Type(false)
{
}

Void::Void(bool isTemp) : Type(isTemp)
{
}

const bool Void::IsPrintable()
{
    return false;
}

string Void::ToString()
{
    return string();
}

Void* Void::Clone()
{
    return new Void(this->_isTemp);
}

string Void::GetType()
{
    return "<type \'void\'>";
}
